Not being able to open compressed file from 3.2.2 in 3.3 RC7

Reported by @touficbatache in #1449

Confirmed same symptom with current master.

Logcat shown on Oneplus 2 running Slim7 (7.1.2) when trying to open the zero byte file in zip:
```
java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$3.done(AsyncTask.java:325)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
        at java.util.concurrent.FutureTask.run(FutureTask.java:242)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:761)
     Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
        at java.util.ArrayList.get(ArrayList.java:411)
        at com.amaze.filemanager.filesystem.compressed.extractcontents.helpers.ZipExtractor.extractWithFilter(ZipExtractor.java:70)
        at com.amaze.filemanager.filesystem.compressed.extractcontents.Extractor.extractFiles(Extractor.java:61)
        at com.amaze.filemanager.asynchronous.services.ExtractService$DoWork.doInBackground(ExtractService.java:291)
        at com.amaze.filemanager.asynchronous.services.ExtractService$DoWork.doInBackground(ExtractService.java:213)
        at android.os.AsyncTask$2.call(AsyncTask.java:305)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
```
then it crashed. App restarts attempted by system led to another exception (not directly related to problem though IMO):
```
    java.lang.RuntimeException: Unable to start service com.amaze.filemanager.asynchronous.services.ExtractService@5a1fcb with null: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.content.Intent.getStringExtra(java.lang.String)' on a null object reference
        at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3383)
        at android.app.ActivityThread.-wrap21(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1612)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6229)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:891)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:781)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.content.Intent.getStringExtra(java.lang.String)' on a null object reference
        at com.amaze.filemanager.asynchronous.services.ExtractService.onStartCommand(ExtractService.java:87)
```

I can successfully reproduce this issue with my local zip files. Here's the log

````
2018-12-05 19:45:51.539 25377-25377/com.amaze.filemanager E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.amaze.filemanager, PID: 25377
    java.lang.RuntimeException: Unable to start service com.amaze.filemanager.asynchronous.services.ExtractService@4f63ea1 with Intent { cmp=com.amaze.filemanager/.asynchronous.services.ExtractService (has extras) }: java.lang.SecurityException: Permission Denial: startForeground from pid=25377, uid=10318 requires android.permission.FOREGROUND_SERVICE
        at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3821)
        at android.app.ActivityThread.access$1800(ActivityThread.java:207)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1779)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6863)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:537)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.SecurityException: Permission Denial: startForeground from pid=25377, uid=10318 requires android.permission.FOREGROUND_SERVICE
        at android.os.Parcel.createException(Parcel.java:1950)
        at android.os.Parcel.readException(Parcel.java:1918)
        at android.os.Parcel.readException(Parcel.java:1868)
        at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:5146)
        at android.app.Service.startForeground(Service.java:695)
        at com.amaze.filemanager.asynchronous.services.ExtractService.onStartCommand(ExtractService.java:123)
        at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3802)
        at android.app.ActivityThread.access$1800(ActivityThread.java:207) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1779) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6863) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:537) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
     Caused by: android.os.RemoteException: Remote stack trace:
        at com.android.server.am.ActivityManagerService.enforcePermission(ActivityManagerService.java:10697)
        at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:1243)
        at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:924)
        at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:22478)
        at android.app.IActivityManager$Stub.onTransact$setServiceForeground$(IActivityManager.java:10580)
````

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -31,6 +31,7 @@
     <uses-permission android:name="com.amaze.cloud.permission.ACCESS_PROVIDER" />
     <uses-permission android:name="android.permission.USE_FINGERPRINT" />
     <uses-permission android:name="android.permission.REQUEST_INSTALL_PACKAGES" />
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
 
     <uses-feature
         android:name="android.hardware.touchscreen"

